mattermost/mattermost-webapp
# [MM-32804] Use npm package for timezones

#### Summary
This updated the timezone selector and label to use a new npm package for the timezone list. This was done to make the list more readable/user friendly. Some things to note:
- I removed the old timezone list from the redux store, I could not see it being used anywhere else so I replaced it with the new one.
- There should be no breaking changes to any users with their existing timezones.
- There are cases where a user may have picked a manual timezone that isn't in the new list (e.g `Israel`). For any edge cases like this you will keep your existing timezone and we will show it as selected in the dropdown but once you change your timezone, it will no longer be selectable/visible.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32804

#### Test Steps
1. Navigate to Account Settings > Display > Timezone.
2. Activate manual timezones and select a new timezone.
3. Expected result: Labels update correctly
4. Send message in channels and ensure the times are matching the timezone.

Repeat those steps a few more times and toggle between automatic and manual timezones. Select a different manual timezone each time.

#### Screenshots
![Screen Shot 2021-08-06 at 2 53 56 PM](https://user-images.githubusercontent.com/14115924/128558666-c153b834-56bc-4ae3-aadf-d9d3a3eac5c1.png)

#### Release Note

```release-note
Updating Account Settings > Display > Timezone to be more user friendly
```
